x/nftmarket/keeper/nft_listing.go

```diff
@@ -614,6 +614,18 @@ func (k Keeper) HandleFullPaymentsPeriodEndings(ctx sdk.Context) {
 				listing.State = types.ListingState_SUCCESSFUL_BID
 			}
 			k.SaveNftListing(ctx, listing)
+
+			// Reset the loan data for a lister
+			// If the bid.PaidAmount is more than loan.Coin.Amount, then just delete the loan data for lister.
+			// Otherwise, subtract bid.PaidAmount from loaning amount
+			loan := k.GetDebtByNft(ctx, listing.IdBytes())
+			if loan.Loan.Amount.LT(bid.PaidAmount) {
+				k.DeleteDebt(ctx, listing.IdBytes())
+			} else {
+				renewedLoanAmount := loan.Loan.Amount.Sub(bid.PaidAmount)
+				loan.Loan.Amount = renewedLoanAmount
+				k.SetDebt(ctx, loan)
+			}
```

Comment:
Isn't it better to remove whole debt outside of loop? (If debt is remaining after full successful bid, it should be something weird and we will need to handle something. Since after successful bid, nft listing should be removed and loan on not available listing would be breaking the system.

Reply:
> Since after successful bid, nft listing should be removed and loan on not available listing would be breaking the system

yes, that's issue #297 .
So, you think we should handle #297 at the same time #295 does?

Reply:
Yes, I think it would be nice to handle these things on a single PR to make sure all logic is accurate. (Probably, it's much better if these could be handled on a single function)

Reply:
okay. understood.
Will consider #297 as a set of #295 for a better solution for those!